webplugin/js/app/km-widget-custom-events.js

```diff
@@ -0,0 +1,179 @@
+    },
+    startConversationEvent: {
+        eventCateogry: 'Kommunicate',
+        eventAction: 'Start New',
+        eventLabel: 'Conversation Start',
+    },
+    greetingMessageEvent: {
+        eventCateogry: 'Kommunicate',
+        eventAction: 'Click',
+        eventLabel: 'Greeting',
+    },
+    conversationRestartEvent: {
+        eventCateogry: 'Kommunicate',
+        eventAction: 'Restart',
+        eventLabel: 'Conversation Restart',
+    },
+    ratingEvent: {
+        eventCateogry: 'Kommunicate',
+        eventAction: 'Rate',
+    },
+    locationEvent: {
+        eventCateogry: 'Kommunicate',
+        eventAction: 'Click',
+        eventLabel: 'Location',
+    },
+    attachmentEvent: {
+        eventCateogry: 'Kommunicate',
+        eventAction: 'Click',
+        eventLabel: 'Attachment',
+    },
+    notificationEvent: {
+        eventCateogry: 'Kommunicate',
+        eventAction: 'Click',
+        eventLabel: 'Notification',
+    },
+    voiceInputEvent: {
+        eventCateogry: 'Kommunicate',
+        eventAction: 'Click',
+        eventLabel: 'VoiceInput',
+    },
+};
+
+kmWidgetEvents = {
+    onChatWidgetOpen: function (resp) {},
+    onChatWidgetClose: function (resp) {},
+    onFaqClick: function (resp) {},
+    onStartNewConversation: function (resp) {},
+    onAttachmentClick: function (resp) {},
+    onVoiceIconClick: function (resp) {},
+    onLocationIconClick: function (resp) {},
+    onRateConversationClick: function (resp) {},
+    onRateConversationEmoticonsClick: function (resp) {},
+    onSubmitRatingClick: function (resp) {},
+    onRestartConversationClick: function (resp) {},
+    onGreetingMessageNotificationClick: function (resp) {},
+    onNotificationClick: function (resp) {},
+    onRichMessageButtonClick: function (resp) {},
+    onShowResolvedClick: function (resp) {},
+    sendEventToGoogleAnalytics: function (
+        eventCateogry,
+        eventAction,
+        eventLabel,
+        eventValue
+    ) {
+        var trackingID = applozic._globals.gaTrackingID;
+        if (trackingID && typeof window.top.ga !== 'undefined') {
+            window.top.ga('create', trackingID.toString(), 'auto');
+            window.top.ga('send', {
+                hitType: 'event',
+                eventCategory: eventCateogry,
+                eventAction: eventAction,
+                eventLabel: eventLabel,
+                eventValue: eventValue,
+            });
+        }
+    },
+    eventTracking: function (eventObject) {
+        if (kommunicateCommons.isObject(eventObject)) {
+            applozic._globals.gaTrackingID &&
+                kmWidgetEvents.sendEventToGoogleAnalytics(
+                    eventObject.eventCateogry,
+                    eventObject.eventAction,
+                    eventObject.eventLabel,
+                    eventObject.eventValue
+                );
+            if (eventObject.eventLabel == 'FAQ Menu') {
+                window.kmWidgetEvents.onFaqClick(eventObject);
```

Comment:
what if we change
faqEvent: {
        eventCateogry: 'Kommunicate',
        eventAction: 'Click',
        eventLabel: 'FAQ Menu',
    }

to include the function name eg:

faqEvent: {
       data: {      
       eventCateogry: 'Kommunicate',
        eventAction: 'Click',
        eventLabel: 'FAQ Menu'
},
        eventFunc: window.kmWidgetEvents.onFaqClick
    }

and then here instead of multiple if condition for each event,
code we can write as:
eventObject.eventFunc(eventObject.data);

Reply:
In that case, if we need to call for every `eventMapping.faqEvent.eventFunc(eventObject.data);` in its respective function (where the base function is executing for the particular functionality) execution, can we do it like `kmWidgetEvents.onFaqClick(faqEvent.data)`?

Reply:
@devashishmamgain  pls refer to the above comment.

Reply:
it will be eventObject.eventFunc(eventObject.data)
and as eventFunc already have reference to onFaqClick and respective function, it will work by above line